### PR TITLE
Add an option to display the result count on a listing view.

### PR DESCRIPTION
**Pull request by instification**
Pretty useful addition for getting the total number of items being displayed.

### CODE REVIEW

File: src/collective/listingviews/browser/views/templates/layout.pt [hunk 1]
```diff
@@ -14,6 +14,9 @@
 
         <div class="listingviews" tal:define="adapter nocall:context/@@listing_view_adapter;
             batch folderContents | adapter/retrieve_listing_items;">
+
+             <p id="result_count" tal:condition="python:adapter.display_count">Showing <span tal:content="adapter/number_of_items" /> items</p>
```
Comment (djay) [inline src/collective/listingviews/browser/views/templates/layout.pt]: no i18n

Comment (instification) [reply src/collective/listingviews/browser/views/templates/layout.pt]: What would i18n look like here?

Doesn't seem to be any internationalisation on the package, so perhaps this is more something for a separate issue/PR?

If it's just a case of adding the i18n schema declaration and `i18n:domain="collective.listingviews"` to the paragraph then easily done in this PR.

**Comment by djay** [comment]
Yes just add it for this

File: src/collective/listingviews/browser/views/templates/layout.pt [hunk 2]
```diff
@@ -14,6 +14,9 @@
 
         <div class="listingviews" tal:define="adapter nocall:view/listing_view_adapter;
             batch folderContents | adapter/retrieve_listing_items;">
+
+             <p id="result_count" tal:condition="python:adapter.display_count"><tal:translate i18n:translate="result_count_showing">Showing</tal:translate><tal:translate i18n:translate="result_count"> <span tal:content="adapter/number_of_items" /> </tal:translate><tal:translate i18n:translate="result_count_items">items</tal:translate></p>
```
Comment (djay) [inline src/collective/listingviews/browser/views/templates/layout.pt]: div is better and I think for the purposes of diazo you need to seperate the label from the number with classes so you can customise it

Comment (djay) [reply src/collective/listingviews/browser/views/templates/layout.pt]: are you sure the translation works that way with the dynamic number inside it?

**Comment by djay** [comment]
need to fix tests

**Comment by djay** [comment]
@instification I'm changing it to reuse plones search page text. ie ```<span class="listing-results-count"><strong class="listing-results-number">3</strong> items matching your search terms.</span>```

Not ideal for all situations but I figure most of the time you want this when used in conjunction with c.collectionfilter or where you have different criteria. If just listings then the batching should be ok. You can always use diazo or translation to change it

**Comment by instification** [comment]
@djay I've fixed the failing tests. 

There is a weird scenario in 4.1 where collections don't include the parent folder but 4.2 onwards does.

As far as I can see this PR is now good to go!

**Pull request merged**